amazon/ion/core.py

```diff
@@ -630,6 +633,15 @@ def __iter__(self):
         for key in six.iterkeys(self.__store):
             yield key
 
+    if _PY_VERSION >= 3:
```

Comment:
Instead of only defining `__str__` for Python 3+, we should try to make it work for Python 2 as well. `six` should be able to provide an abstraction that you can use.